examples/arc-brushing/scatterplot-brushing-layer/scatterplot-brushing-layer-vertex.glsl.js

```diff
@@ -0,0 +1,110 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+export default `\
+#define SHADER_NAME scatterplot-brushing-ayer-vertex-shader
+const float R_EARTH = 6371000.; // earth radius in km
+
+attribute vec3 positions;
+
+attribute vec3 instancePositions;
+attribute vec3 instanceTargetPositions;
+attribute float instanceRadius;
+attribute vec4 instanceColors;
+attribute vec3 instancePickingColors;
+
+uniform float opacity;
+uniform float radiusScale;
+uniform float radiusMinPixels;
+uniform float radiusMaxPixels;
+uniform float renderPickingBuffer;
+uniform float outline;
+uniform float strokeWidth;
+
+// uniform for brushing
+uniform vec2 mousePos;
+uniform float brushRadius;
+uniform float enableBrushing;
+uniform float brushTarget;
+
+varying vec4 vColor;
+varying vec2 unitPosition;
+varying float innerUnitRadius;
+
+// approximate distance between lng lat in meters
+float distanceBetweenLatLng(vec2 source, vec2 target) {
+
+  vec2 delta = (source - target) * PI / 180.;
+
+  float a =
+    sin(delta.y / 2.) * sin(delta.y / 2.) +
+    cos(source.y * PI / 180.) * cos(target.y * PI / 180.) *
+    sin(delta.x / 2.) * sin(delta.x / 2.);
+
+  float c = 2. * atan(sqrt(a), sqrt(1. - a));
+
+  return R_EARTH * c;
+}
+
+// range is km
+float isPointInRange(vec2 ptLatLng, vec2 mouseLatLng, float range, float enabled) {
+
+  return float(enabled <= 0. || distanceBetweenLatLng(ptLatLng, mouseLatLng) <= range);
+}
+
+void main(void) {
+
+  // if enableBrushing is truthy calculate whether instancePosition is in range
+  float isPtInBrush = isPointInRange(instancePositions.xy, mousePos, brushRadius, enableBrushing);
+
+  // for use with arc layer, if brushTarget is truthy 
+  // calculate whether instanceTargetPositions is in range
+  float isTargetInBrush = isPointInRange(instanceTargetPositions.xy, mousePos, brushRadius, 1.);
+
+  float isInBrush = float(float(isPtInBrush > 0. && brushTarget <= 0.) > 0. || 
```

Comment:
... this boolean statement looks rather complicated to me... break them down or put some comments?